klippy/extras/tmc2209.py

```diff
@@ -0,0 +1,92 @@
+# TMC2209 configuration
+#
+# Copyright (C) 2019  Stephan Oelze stephan.oelze@gmail.com
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import tmc2208, tmc2130, tmc, tmc_uart
+TMC_FREQUENCY=12000000.
+
+Registers = dict(tmc2208.Registers)
+
+Registers.update({
+    "TCOOLTHRS": 0x14,
+    "COOLCONF": 0x42,
+    "SGTHRS": 0x40
+})
+
+ReadRegisters = tmc2208.ReadRegisters
+
+fields = dict(tmc2208.Fields)
+fields["COOLCONF"] = {
+    "semin":                    0x0F << 0,
+    "seup":                     0x03 << 5,
+    "semax":                    0x0F << 8,
+    "sedn":                     0x03 << 13,
+    "seimin":                   0x01 << 15
+}
+fields["SGTHRS"] = {
+    "sgt": 0xFF << 0
+}
+
+FieldFormatters = dict(tmc2208.FieldFormatters)
+
+#
+
+######################################################################
+# TMC2209 printer object
+######################################################################
+
+class TMC2209:
+    def __init__(self, config):
+        # Setup mcu communication
+        self.fields = tmc.FieldHelper(fields, tmc2208.SignedFields,\
+            FieldFormatters)
+        self.mcu_tmc = tmc_uart.MCU_TMC_uart(config, Registers, self.fields)
+        # Allow virtual endstop to be created
+        diag1_pin = config.get('diag1_pin', None)
```

Comment:
The tmc2209 doesn't have a "diag1_pin" - it has a "diag_pin".

Reply:
fixed